_posts/2022-05-13-sending-logs-metrics-to-aws.markdown

```diff
@@ -0,0 +1,412 @@
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: logstash-syslog
+  namespace: default
+data:
+  Address: 192.168.1.134:5140
+  Protocol: tcp
+```
+
+Here we use the cluster host's IP where logstash is running. The service will be reachable thanks to the `port-forward` command previously executed on the cluster's host.
+
+Next is to update the EdgeDevice specification to expose the log collector. Here we define the log collector `logstash-syslog` with a maximum buffer size of 10mb. 
+
+```yaml
+spec:
+  logCollection:
+    logstash-syslog:
+      bufferSize: 10
+      kind: syslog
+      syslogConfig:
+        name: logstash-syslog
+```
+
+To instruct the workload to use the given log collector, we just have to define it as part of the manifest:
+
+```yaml
+apiVersion: management.project-flotta.io/v1alpha1
+kind: EdgeWorkload
+metadata:
+  name: random-workload
+spec:
+  logCollection: logstash-syslog
+  deviceSelector:
+    matchLabels:
+      app: foo
+  type: pod
+  pod:
+    spec:
+      containers:
+        - name: random-server
+          image: docker.io/eloycoto/logexample
+```
+
+
+And with that, the logs will be sent to our backend in AWS. To confirm that it's working fine, we query the service to list the contents of the index we defined in the configmap: 
+```
+index => "logstash-%{+YYYY.MM.dd}"
+```
+
+Which, in this case it translates to `logstash-2022.05.12`:
+
+```bash
+username=$(oc get secret logstash-secret -o jsonpath='{.data.LOGSTASH_USERNAME}' | base64 -d)
+password=$(oc get secret logstash-secret -o jsonpath='{.data.LOGSTASH_PASSWORD}' | base64 -d)
+curl -u $username:$password -X GET "https://search-project-flotta-lpfibcnqkcpgavdbprrueysk6a.us-east-1.es.amazonaws.com:443/logstash-2022.05.12/_search?pretty=true" -H 'Content-Type: application/json' -d'
+{
+    "query": {
+        "match_all": {}
+    }
+}
+'
+```
+
+And we get a few results already, displaying the first one to reduce cluttery. 
+
+```json
+  "hits" : {
+    "total" : {
+      "value" : 1162,
+      "relation" : "eq"
+    },
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "logstash-2022.05.12",
+        "_type" : "_doc",
+        "_id" : "z07vuYAB7aBvPfPHfBaK",
+        "_score" : 1.0,
+        "_source" : {
+          "@version" : "1",
+          "severity" : 0,
+          "facility" : 0,
+          "severity_label" : "Emergency",
+          "@timestamp" : "2022-05-12T20:22:01.494Z",
+          "host" : "127.0.0.1",
+          "tags" : [
+            "_grokparsefailure_sysloginput"
+          ],
+          "facility_label" : "kernel",
+          "priority" : 0,
+          "message" : "<6>2022-05-12T16:22:01-04:00  random-workload[3966]: dcc80a53ab7e5871f38e6b5184d785d58f5c698505fe74dd04597dcaac372c1b: New message at: Thu May 12 20:22:01 UTC 2022\n"
+        }
+      },
+...
+```
+
+
+** Setting up prometheus adapter for AWS Timestream **
```

Comment:
This format is not working at all, but maybe ### Setting up prometheus adapter for AWS Timestream

Reply:
Got it, I'll see if I can fix that and use the `###` markdown for header.